src/node_url.cc

```diff
@@ -2036,7 +2036,7 @@ void URL::Parse(const char* input,
               (ch == kEOL ||
                ch == '?' ||
                ch == '#')) {
-            while (url->path.size() > 1 && url->path[0].length() == 0) {
+            while (url->path.size() > 1 && url->path[0].empty()) {
               url->path.erase(url->path.begin());
             }
```

Comment:
Wildly inefficient way of removing empty prefixes... (_n_ reallocations + _n*m_ copies when 1 + _m_ suffices) 
```suggestion
            auto begin = url->path.begin();
            auto end = url->path.end();
            auto from = begin;
            auto to = begin;
            while (to != end && to->empty()) ++to;
            if (to == end && from == begin && from != to) ++from;
            url->path.erase(from, to);  // 1 + m
```

Reply:
@bnoordhuis In your suggestion, the `from == begin` part is always true, unless i’m missing something?

Reply:
Sorry, yeah. It was intended as a kind of self-documentation about the preconditions but I guess it's confusing more than anything else. `if (to == end && from != end) ++from;` is probably a clearer way of writing it.

Reply:
As this is not directly related to the other changes in this PR, this could be done in a separate commit/PR.